apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -225,11 +225,128 @@ type PreferredClusterSelector struct {
 	Preference ClusterSelectorTerm `json:"preference"`
 }
 
+// InterpolationOrder is the order where Fleet should perform linear interpolation
+// for weights in a preferred range.
+// +enum
+type InterpolationOrder string
+
+const (
+	// FromMinimum instructs Fleet to perform linear interpolation of weights starting
+	// from the minimum value, i.e., when the minimum value is most preferred.
+	//
+	// For example, for a preferred range [10, 30] with a minimum weight of 10 and a maximum
+	// weight of 100, Fleet will assign:
+	// * a weight of 100 to the cluster with the minimum value in the range; and
+	// * a weight of 10 to the cluster with the maximum value in the range; and
+	// * a weight of 37 to the cluster with a value of 16 in the range.
+	//
+	//   It is calculated using the formula below:
+	//   100-(16-10)/(30-10)*(100-10) = 73
+	FromMinimum InterpolationOrder = "FromMinimum"
+	// FromMaximum instructs Fleet to perform linear interpolation of weights starting
+	// from the maximum value, i.e., when the maximum value is most preferred.
+	//
+	// For example, for a preferred range [10, 30] with a minimum weight of 10 and a maximum
+	// weight of 100, Fleet will assign:
+	// * a weight of 10 to the cluster with the minimum value in the range; and
+	// * a weight of 100 to the cluster with the maximum value in the range; and
+	// * a weight of 37 to the cluster with a value of 16 in the range.
+	//
+	//   It is calculated using the formula below:
+	//   10+(16-10)/(30-10)*(100-10) = 37
+	FromMaximum InterpolationOrder = "FromMaximum"
+	// DoNotInterpolate instructs Fleet not to perform linear interpolation; when this
+	// option is used, alls clusters with a value in the preferred range will be assigned
+	// the same weight.
+	//
+	// This is the default option.
+	DoNotInterpolate InterpolationOrder = "DoNotInterpolate"
+)
+
+// MetricRange describes a required or preferred range for a metric.
+type MetricRange struct {
+	// Minimum is the minimum value of the range.
+	// If this value is not specified, the smallest observed value of the metric will
+	// be used.
+	// At least one of the minimum or maximum value must be specified.
+	// +optional
+	Minimum string `json:"minimum"`
+	// Maximum is the maximum value of the range.
```

Comment:
nit: we keep an empty line between fields

Reply:
Sure, will update.